tests/llmobs/test_llmobs.py

```diff
@@ -0,0 +1,253 @@
+            with tracer.trace("child_span"):
+                with tracer.trace("llm_grandchild_span", span_type=SpanTypes.LLM) as grandchild_span:
+                    grandchild_span._set_ctx_item(const.SPAN_KIND, "llm")
+                    with tracer.trace("great_grandchild_span", span_type=SpanTypes.LLM) as great_grandchild_span:
+                        great_grandchild_span._set_ctx_item(const.SPAN_KIND, "llm")
+
+        llm_event, grandchild_event, great_grandchild_event = llmobs_events
+        assert llm_event["session_id"] == "session-123"
+        assert grandchild_event["session_id"] == "session-123"
+        assert great_grandchild_event["session_id"] == "session-123"
+
+
+def test_input_value_is_set(tracer, llmobs_events):
+    """Test that input value is set on the span event if they are present on the span."""
+    with tracer.trace("root_llm_span", span_type=SpanTypes.LLM) as llm_span:
+        llm_span._set_ctx_item(const.SPAN_KIND, "llm")
+        llm_span._set_ctx_item(const.INPUT_VALUE, "value")
+    assert llmobs_events[0]["meta"]["input"]["value"] == "value"
+
+
+def test_input_messages_are_set(tracer, llmobs_events):
+    """Test that input messages are set on the span event if they are present on the span."""
+    with tracer.trace("root_llm_span", span_type=SpanTypes.LLM) as llm_span:
+        llm_span._set_ctx_item(const.SPAN_KIND, "llm")
+        llm_span._set_ctx_item(const.INPUT_MESSAGES, [{"content": "message", "role": "user"}])
+    assert llmobs_events[0]["meta"]["input"]["messages"] == [{"content": "message", "role": "user"}]
+
+
+def test_input_parameters_are_set(tracer, llmobs_events):
+    """Test that input parameters are set on the span event if they are present on the span."""
+    with tracer.trace("root_llm_span", span_type=SpanTypes.LLM) as llm_span:
+        llm_span._set_ctx_item(const.SPAN_KIND, "llm")
+        llm_span._set_ctx_item(const.INPUT_PARAMETERS, {"key": "value"})
+    assert llmobs_events[0]["meta"]["input"]["parameters"] == {"key": "value"}
+
+
+def test_output_messages_are_set(tracer, llmobs_events):
+    """Test that output messages are set on the span event if they are present on the span."""
+    with tracer.trace("root_llm_span", span_type=SpanTypes.LLM) as llm_span:
+        llm_span._set_ctx_item(const.SPAN_KIND, "llm")
+        llm_span._set_ctx_item(const.OUTPUT_MESSAGES, [{"content": "message", "role": "user"}])
+    assert llmobs_events[0]["meta"]["output"]["messages"] == [{"content": "message", "role": "user"}]
+
+    def test_output_value_is_set(self, tracer, llmobs_events):
+        """Test that output value is set on the span event if they are present on the span."""
+        with tracer.trace("root_llm_span", span_type=SpanTypes.LLM) as llm_span:
+            llm_span._set_ctx_item(const.SPAN_KIND, "llm")
+            llm_span._set_ctx_item(const.OUTPUT_VALUE, "value")
+        assert llmobs_events[0]["meta"]["output"]["value"] == "value"
```

Comment:
Think this needs to be unindented

Reply:
if it's unindented then the trace isn't finished and so no TraceProcessor invocation and no events generated

Reply:
Oh no I meant I think the test needs to be unindented, it's part of the previous test lol

Reply:
oh lol 🤦‍♂️ great catch lmao

Reply:
done in [03af86c](https://github.com/DataDog/dd-trace-py/pull/11784/commits/03af86c60018e5a84a320806949c28ad9569d95e)